yt/frontends/parthenon/data_structures.py

```diff
@@ -0,0 +1,321 @@
+import os
+import warnings
+import weakref
+from itertools import chain, product
+
+import numpy as np
+
+from yt.data_objects.index_subobjects.grid_patch import AMRGridPatch
+from yt.data_objects.static_output import Dataset
+from yt.fields.magnetic_field import get_magnetic_normalization
+from yt.funcs import mylog
+from yt.geometry.geometry_enum import Geometry
+from yt.geometry.grid_geometry_handler import GridIndex
+from yt.utilities.chemical_formulas import compute_mu
+from yt.utilities.file_handler import HDF5FileHandler
+
+from .fields import ParthenonFieldInfo
+
+geom_map = {
+    "UniformCartesian": Geometry.CARTESIAN,
+}
+
+_cis = np.fromiter(
+    chain.from_iterable(product([0, 1], [0, 1], [0, 1])), dtype=np.int64, count=8 * 3
+)
+_cis.shape = (8, 3)
+
+
+class ParthenonGrid(AMRGridPatch):
+    _id_offset = 0
+
+    def __init__(self, id, index, level):
+        AMRGridPatch.__init__(self, id, filename=index.index_filename, index=index)
+        self.Parent = None
+        self.Children = []
+        self.Level = level
+
+    def _setup_dx(self):
+        # So first we figure out what the index is.  We don't assume
+        # that dx=dy=dz , at least here.  We probably do elsewhere.
+        id = self.id - self._id_offset
+        LE, RE = self.index.grid_left_edge[id, :], self.index.grid_right_edge[id, :]
+        self.dds = self.ds.arr((RE - LE) / self.ActiveDimensions, "code_length")
+        if self.ds.dimensionality < 2:
+            self.dds[1] = 1.0
+        if self.ds.dimensionality < 3:
+            self.dds[2] = 1.0
+        self.field_data["dx"], self.field_data["dy"], self.field_data["dz"] = self.dds
+
+    def retrieve_ghost_zones(self, n_zones, fields, all_levels=False, smoothed=False):
+        if smoothed:
+            warnings.warn(
+                "ghost-zones interpolation/smoothing is not "
+                "currently supported for Parthenon data.",
+                category=RuntimeWarning,
```

Comment:
`ruff` was recently updated and now points to this warning not having an explicit `stack_level` arg. In general is should be
```suggestion
                category=RuntimeWarning,
                stack_level=2,
```